compiler/plc_xml/src/model/fbd.rs

```diff
@@ -105,14 +122,239 @@ impl Parseable for FunctionBlockDiagram {
         }
 
         nodes.sort_by(|_, b, _, d| b.partial_cmp(d).unwrap()); // This _shouldn't_ panic because our `partial_cmp` method covers all cases
+
         Ok(FunctionBlockDiagram { nodes })
     }
 }
 
+enum SourceReference {
+    Value(NodeId),
+    Connector(NodeId),
+    Unconnected(NodeId),
+}
+
+struct ResolvedConnection {
+    id: NodeId,
+    resolved_ref_id: NodeId,
+    block_parameter_index: Option<usize>,
+}
+
+// IndexMap<NodeId, Node> interface for connection-point (sink/source) desugaring
+trait ConnectionResolver<'xml> {
```

Comment:
I'd like to have isolated unit tests for this functionality. 
We should also document what's happening more, and maybe have a walk-through on how this happens.
if you're up for it we could try to do it together today or tomorrow.